src/main/java/de/tum/cit/aet/artemis/atlas/service/learningpath/LearningPathRecommendationService.java

```diff
@@ -489,11 +489,11 @@ public List<LearningObject> getRecommendedOrderOfLearningObjects(User user, Cour
 
         final var numberOfRequiredExercisePointsToMaster = calculateNumberOfExercisePointsRequiredToMaster(user, competency, weightedConfidence);
 
-        final var pendingExercises = competency.getExerciseLinks().stream().map(CompetencyExerciseLink::getExercise)
-                .filter(exercise -> !learningObjectService.isCompletedByUser(exercise, user)).collect(Collectors.toSet());
+        final var pendingExercises = competency.getExerciseLinks().stream().sorted(Comparator.comparingDouble(CompetencyExerciseLink::getWeight).reversed())
```

Comment:
Why do we reverse the list before filtering? 
Why is the reversing needed at all? Wouldn't it be more efficient to do it after filtering?

Reply:
The list is not reversed, only the comparator. This is done to get the highest linked exercises first instead of last (normal sorting).
But I agree about the filtering part, changed!